web/src/beta/features/Editor/tabs/map/RightPanel/index.tsx

```diff
@@ -46,7 +48,10 @@ const MapRightPanel: React.FC<Props> = ({
   const { scene } = useSceneQuery({ sceneId });
 
   const scenePropertyId = useMemo(() => scene?.property?.id, [scene?.property?.id]);
-  const sceneSettings = useMemo(() => convert(scene?.property), [scene?.property]);
+  const sceneSettings = useMemo(
+    () => convert(scene?.property)?.filter(item => item.collection === selectedSceneSetting),
```

Comment:
Can you add the converting of the scene property to inside the scene API and update the scene type there?
Currently we are asserting the type as the GQL type, which isn't great, so if you can update the type too that would be really helpful 🙏🏻

Reply:
Of course the filtering here should stay here

Reply:
sorry would you explain more , you mean move this convert to useScene, not scene Api right? and this scene setting have the type Item not GQL type . is that the type you are talking about?

Reply:
If you check other useXxxxApi's and how they are doing properties, you will see they should all be doing `convert(asldkfj?.property)` there.
Just want to keep it consistent. But this also means you'll need to update/add a Scene type there

Reply:
I apply your suggestion ,but I didn't need to define type , but can take a look is that what you mean ?